lib/internal/Magento/Framework/GetParameterClassTrait.php

```diff
@@ -0,0 +1,37 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+namespace Magento\Framework;
+
+use ReflectionClass;
+use ReflectionParameter;
+
+/**
+ * Returns a reflection parameter's class if possible.
+ */
+trait GetParameterClassTrait
+{
+    /**
+     * Get class by reflection parameter
+     *
+     * @param ReflectionParameter $reflectionParameter
+     *
+     * @return ReflectionClass|null
+     * @throws ReflectionException
+     */
+    private function getParameterClass(ReflectionParameter $reflectionParameter): ?ReflectionClass
+    {
+        $parameterType = $reflectionParameter->getType();
+        // In PHP8, $parameterType could be an instance of ReflectionUnionType, which doesn't have isBuiltin method.
+        if ($parameterType !== null && method_exists($parameterType, 'isBuiltin') === false) {
```

Comment:
We need more testing here. While magento code will continue to use PHP7.4 syntax we should be able to work with code which is PHP8 specific. In future someone may develop extension with PHP8 as minimal version(and PHP8 syntax in it). Magento should properly work with such extensions.
Please explore dev/tests/integration/testsuite/Magento/Framework/ObjectManager to see if we can add tests that will be executed only with PHP8. Would be great to see that magento can properly work with code with union types (i.e generate code, instantiate objects etc)